test/tests.zig

```diff
@@ -1167,6 +1167,100 @@ pub fn addCliTests(b: *std.Build) *Step {
         step.dependOn(&cleanup.step);
     }
 
+    {
+        // Test `zig build -fallow-deprecated`.
+
+        const deprecated_check: std.Build.Step.Run.StdIo.Check = .{
+            .expect_stderr_match = "found deprecated code",
+        };
+
+        const tmp_path = b.makeTempPath();
+
+        // create custom main.zig file containing a deprecated decl
+        {
+            const new_main_src =
+                \\const bad = @deprecated(42);
+                \\
+                \\pub fn main() u8 {
+                \\   return bad;
+                \\}
+                \\
+                \\test {
+                \\   if (bad != 42) return error.Bad;   
+                \\}
+            ;
+
+            var src_dir = std.fs.cwd().makeOpenPath(b.pathJoin(&.{ tmp_path, "src" }), .{}) catch unreachable;
+            defer src_dir.close();
+
+            var main = src_dir.createFile("main.zig", .{}) catch unreachable;
+            defer main.close();
+
+            main.writeAll(new_main_src) catch unreachable;
+        }
```

Comment:
This isn't your fault, since you've just taken it from some of the surrounding cases, but... ooh, using `std.fs` in a build script is very naughty. I'm not gonna ask you to change this for now, though, since you're just following the surrounding cases.

However, these should be `@panic`s, not `unreachable`; please do fix that.